tools/bin/mtest

```diff
@@ -498,73 +530,96 @@ def configure_options(target):
+            # date they were first created instead
+            for config in [new_config, new_crypto_config]:
+                if not os.path.exists(config):
+                    hash = "2f1ae5a86ebd67faa50c3983b4567656c2234674"
+                    log(f"git show {hash}:{config} > {config}")
+                    output, _ = subrun(["git", "show", f"{hash}:{config}"], silent=True)
+                    with open(f"{config}", 'w') as f:
                         f.write(output)
-        elif config.endswith(".h"):
-            # a config file - overwrite existing config file
-            path = config if os.path.exists(config) else f"{ROOT}/configs/{config}"
-            if path.split("/")[-1].startswith("crypto_config_"):
-                dest = CRYPTO_CONFIG_FILE
-            else:
-                dest = CONFIG_FILE
-            log(f"cp {path[len(ROOT) + 1:] if path.startswith(ROOT) else path} {dest[len(ROOT) + 1:]}")
-            shutil.copy(path, dest)
+
+    for config in header_options:
+        # a config file - overwrite existing config file
+        path = config if os.path.exists(config) else f"configs/{config}"
+        if path.split("/")[-1].startswith("crypto_config_"):
+            new_crypto_config = path
         else:
-            # a configuration recognised by config.py, e.g. "baremetal"
-            log(f"scripts/config.py {config}")
-            subrun([f"{ROOT}/scripts/config.py", config], silent=True)
-
-    if disable_spm:
-        # at some point in the past, MBEDTLS_HAVE_ASM also needs disabling,
-        # and more options again (not sure which) for 3.0.0
-        set_config_option("unset", "MBEDTLS_PSA_CRYPTO_SPM", silent=False)
-    if set_no_platform_entropy:
-        # this gets disabled by #8164 - ensure it is set
-        set_config_option("set", "MBEDTLS_NO_PLATFORM_ENTROPY", silent=False)
+            new_config = path
+        if any(x[0] == os.path.basename(path) for x in ["config-tfm.h", "tfm_mbedcrypto_config_profile_medium", "crypto_config_profile_medium.h"]):
+            config_is_tfm = True
+
+    # backup and overwrite the config files
+    if new_config is not None:
+        backup_config_options(CONFIG_FILE)
+        log(f"cp {new_config} {CONFIG_FILE}")
+        shutil.copy(new_config, CONFIG_FILE)
+    if new_crypto_config is not None:
+        crypto_config_file = get_crypto_config_file()
+        backup_config_options(crypto_config_file)
+        log(f"cp {new_crypto_config} {crypto_config_file}")
+        shutil.copy(new_crypto_config, crypto_config_file)
+    
+    sync()
+
+    # apply bulk config.py settings to main config file
+    for config in config_py_groups:
+        options.remove(config)
+        # a configuration recognised by config.py, e.g. "baremetal"
+        backup_config_options(CONFIG_FILE)
+        log(f"scripts/config.py {config}")
+        subrun([f"scripts/config.py", config], silent=True)
+        
+    if config_is_tfm:
```

Comment:
I note that in commit `2f1ae5a86ebd67faa50c3983b4567656c2234674` there is

```
+ * Note that while this define has been removed from TF-M's copy of this config                                                                     
+ * file, TF-M still passes this option to Mbed TLS during the build via CMake.                                                                      
+ * Therefore we keep it in our copy. See discussion on PR #7426 for more info.                                                                      
+ *                                                                                                                                                  
  */                                                                                                                                                 
 #define MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER
```

but we don't have `MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER` anywhere in `configs/`

(capturing here - we may actually want to fix that in the Mbed TLS tree)

Reply:
I think raise this as a separate issue and cc the TF-M people on it.